web-dapp/src/components/App.js

```diff
@@ -21,21 +23,67 @@ class App extends Component {
             contract: null,
         };
         this.check_web3 = this.check_web3.bind(this);
+        this.check_contract = this.check_contract.bind(this);
+    }
+
+    check_contract(cconf, callback) {
+        console.log('checking contract');
+        var contract = window.my_web3.eth.contract(cconf.abi).at(cconf.address);
+        window.megaContract = contract;
+
+        window.my_web3.eth.getCode(cconf.address, function (err, code) {
+            if (err) {
+                console.log('error fetching code from address: ' + cconf.address + ', err: ' + err);
+                return callback();
+            }
+
+            // code here starts with 0x prefix in contrast with cconf.bytecode
+            if (code.length <= 2) {
+                console.log('code at address: ' + cconf.address + ' is empty: ' + code);
+                return callback();
+            }
+
+            contract.owner(function (err, owner) {
+                if (err) {
+                    console.log('error fetching owner: ' + err);
+                    return callback();
+                }
+
+                if (!owner || owner.length <= 2) {
+                    console.log('owner is empty: ' + owner);
+                    return callback();
+                }
+
+                return callback( contract );
+            });
+        });
     }
 
     check_web3() {
         console.log('check_web3');
         if (window.my_web3) {
             console.log('web3 found, getting contract');
             clearInterval(this.state.web3_checker);
-            var cconf = require('../contract-output').ProofOfPhysicalAddress;
-            var contract = window.my_web3.eth.contract(cconf.abi).at(cconf.address);
-            window.megaContract = contract;
-            this.setState({
-                contract: contract,
-                my_web3: window.my_web3,
-                web3_checker: null,
-                web3_checker_dur: 0,
+            var cconf = ContractOutput.ProofOfPhysicalAddress;
```

Comment:
We can solve this issue, at least for now, moving the line 9 to here, and making and require instead of an import. In a next PR @sistemico will add a solve for this.

Reply:
Please feel free to merge this PR to `coverage-web-dapp` branch (instead of `master`) and I will take care of it when merging all the changes needed to increase coverage. I already have an approach to this issue.